stubs/PyYAML/yaml/composer.pyi

```diff
@@ -1,17 +1,18 @@
-from typing import Any
+from typing import Any, Union
 
 from yaml.error import MarkedYAMLError
+from yaml.nodes import Node
 
 class ComposerError(MarkedYAMLError): ...
 
 class Composer:
     anchors: Any
     def __init__(self) -> None: ...
-    def check_node(self): ...
-    def get_node(self): ...
-    def get_single_node(self): ...
-    def compose_document(self): ...
-    def compose_node(self, parent, index): ...
-    def compose_scalar_node(self, anchor): ...
-    def compose_sequence_node(self, anchor): ...
-    def compose_mapping_node(self, anchor): ...
+    def check_node(self) -> bool: ...
+    def get_node(self) -> Union[Node, None]: ...
+    def get_single_node(self) -> Union[Node, None]: ...
+    def compose_document(self) -> Union[Node, None]: ...
+    def compose_node(self, parent: Union[Node, None], index: int) -> Union[Node, None]: ...
+    def compose_scalar_node(self, anchor: Any) -> Node: ...
+    def compose_sequence_node(self, anchor: Any) -> Node: ...
+    def compose_mapping_node(self, anchor: Any) -> Node: ...
```

Comment:
Usually more specific return types are better. Also, if you don't know the type of an argument, leave it out instead of writing `: Any`.

```suggestion
    def compose_scalar_node(self, anchor) -> ScalarNode: ...
    def compose_sequence_node(self, anchor) -> SequenceNode: ...
    def compose_mapping_node(self, anchor) -> MappingNode: ...
```

Of course, you will need to import the `FooNode` classes.